language/cc/config.go

```diff
@@ -87,11 +110,16 @@ func newCppConfig() *cppConfig {
 	return &cppConfig{
 		groupingMode:            groupSourcesByDirectory,
 		groupsCycleHandlingMode: mergeOnGroupsCycle,
+		dependencyIndexes:       []ccDependencyIndex{},
 	}
 }
 func (conf *cppConfig) clone() *cppConfig {
-	copy := *conf
-	return &copy
+	return &cppConfig{
+		groupingMode:            conf.groupingMode,
+		groupsCycleHandlingMode: conf.groupsCycleHandlingMode,
+		// No deep cloning of dependency indexes to reduce memory usage
+		dependencyIndexes: conf.dependencyIndexes,
```

Comment:
Prevent a subtle case where `Configure` can overwrite elements of the same backing array in different directories. This effectively makes `dependencyIndexes` copy-on-append.

```suggestion
		dependencyIndexes: conf.dependencyIndexes[:len(conf.dependencyIndexes):len(conf.dependencyIndexes)],
```